File: backend/byteBlog/src/main/java/com/jeswin8801/byteBlog/entities/dto/comment/CommentCompactDto.java
package com.jeswin8801.byteBlog.entities.dto.comment;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CommentCompactDto {
    @JsonProperty("blog-heading-uri")
    private String blogHeadingUri;

    private String comment;

    @JsonProperty("last-updated")
    private Instant lastUpdated; // will be sorted by client

    @JsonProperty("reply-comments-author-profile-image-uris")
    private Set<String> replyCommentAuthorProfileImageUris;
}

File: frontend/src/app/models/dtos/comment/comment-compact-dto.ts
import { JsonProperty } from 'json-object-mapper';

export class CommentCompactDto {
  @JsonProperty({ name: 'blog-heading-uri' })
  blogHeadingUri?: string;

  comment?: string;

  @JsonProperty({ name: 'last-updated' })
  lastUpdated?: string;

  @JsonProperty({ name: 'reply-comments-author-profile-image-uris' })
  replyCommentAuthorProfileImageUris?: string[];
}
